kyuubi-server/src/main/scala/org/apache/kyuubi/engine/EngineRef.scala

```diff
@@ -109,6 +96,47 @@ private[kyuubi] class EngineRef(
     case _ => user
   }
 
+  @VisibleForTesting
+  private[kyuubi] val subdomain: String = conf.get(ENGINE_SHARE_LEVEL_SUBDOMAIN) match {
+    case Some(_subdomain) => _subdomain
+    case None if clientPoolSize > 0 =>
+      val poolSize = math.min(clientPoolSize, poolThreshold)
+      if (poolSize < clientPoolSize) {
+        warn(s"Request engine pool size($clientPoolSize) exceeds, fallback to " +
+          s"system threshold $poolThreshold")
+      }
+      val seqNum = enginePoolBalancePolicy match {
+        case "POLLING" =>
+          info(s"The engine pool balance policy is POLLING.")
+          val snPath =
+            DiscoveryPaths.makePath(
+              s"${serverSpace}_$shareLevel",
+              "seq_num",
+              Array(appUser, clientPoolName))
```

Comment:
@pan3793  do you have some suggestion for the data path?

- it should contains kyuubi version
- it should contains engine type
- share level is not needed, because it is always user share level

Reply:
the engine pool is designed to be orthogonal w/ the engine share level, it says we can combine the GROUP share level w/ the pool feature.

Reply:
changed to             
DiscoveryPaths.makePath(
              s"${serverSpace}_${KYUUBI_VERSION}_${shareLevel}_$engineType",
              "seq_num",
              Array(appUser, clientPoolName))